### PR TITLE
Tms admin cannot modify users #112

### LINKED ISSUE #112: Story 17: Admin cannot modify users
As a logged in Admin
      I can modify my account data
      But I cannot modify any other user's account data

### PR DESCRIPTION
**Pull request by tmikeschu**
Closes #112 

Ready for review, all hands on deck @wlffann @akintner @kheppenstall 

- Add edit functionality to user information

- Add link to dashboard view to update admin profile

- Add verification method to user model

- Add form partial for user forms

- Add role option to user factory

- Add login stub helper method (for when we don't need a full login sequence)

### CODE REVIEW

File: app/views/users/_form.html.haml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      %article.form-field
+        = f.text_field :first_name, hide_label: true, placeholder: 'first_name'
+      %article.form-field
+        = f.text_field :last_name, hide_label: true, placeholder: 'last_name'
+      %article.form-field
+        = f.email_field :email, hide_label: true, placeholder: 'email'
+
+      - if @user.new_record?
+        %article.form-field
+          = f.password_field :password, hide_label: true, placeholder: 'password'
```
Comment (akintner) [inline app/views/users/_form.html.haml]: Liking the hide_label and placeholder for password

**Comment by akintner** [comment]
I like the new form partial. Everything is looking good here.

File: app/models/user.rb [hunk 2]
```diff
@@ -12,4 +12,8 @@ class User < ApplicationRecord
   def name
     "#{first_name} #{last_name}"
   end
+
+  def verified?(params_id)
+    id == params_id.to_i
```
Comment (wlffann) [inline app/models/user.rb]: Love this!

File: spec/support/feature_helpers.rb [hunk 3]
```diff
@@ -67,4 +67,8 @@ def expect_item_info(item, quantity)
   def total(item1, item2, quantity = 1)
     (item1.price_per_unit * quantity + item2.price_per_unit).round(2)
   end
+
+  def login_stub(user)
```
Comment (wlffann) [inline spec/support/feature_helpers.rb]: Super helpful, thanks!

**Comment by kheppenstall** [comment]
👍 Good to go.

**Pull request merged**